File: Sciencific_Article/Sciencific_Article.Domain/Entities/Role.cs
using System;
using System.Collections.Generic;

namespace Sciencific_Article.Domain.Entities;

public partial class Role
{
    public string RoleId { get; set; } = null!;

    public string? RoleName { get; set; }

    public string? Description { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<User> Users { get; set; } = new List<User>();
}

File: Journal_Trend_Analysis/lib/models/role.dart
class Role {
  final String roleId;
  final String? roleName;
  final String? description;

  const Role({
    required this.roleId,
    this.roleName,
    this.description,
  });

  factory Role.fromJson(Map<String, dynamic> json) {
    return Role(
      roleId: json['roleId']?.toString() ?? json['role_id']?.toString() ?? '',
      roleName: json['roleName']?.toString() ?? json['role_name']?.toString(),
      description: json['description']?.toString(),
    );
  }
}
